packages/core/test/interpreter.test.ts

```diff
@@ -1847,4 +1778,115 @@ describe('interpreter', () => {
       });
     });
   });
+
+  describe('fromPromise', () => {
+    it('should resolve', (done) => {
+      const actor = interpret(fromPromise(() => Promise.resolve(42)));
+
+      actor.subscribe((state) => {
+        if (state === 42) {
+          done();
+        }
+      });
+
+      actor.start();
+    });
+
+    it('should resolve (observer .next)', (done) => {
+      const actor = interpret(fromPromise(() => Promise.resolve(42)));
+
+      actor.subscribe({
+        next: (state) => {
+          if (state === 42) {
+            done();
+          }
+        }
+      });
+
+      actor.start();
+    });
+
+    it('should reject (observer .error)', (done) => {
+      const actor = interpret(fromPromise(() => Promise.reject('Error')));
+
+      actor.subscribe({
+        error: (data) => {
+          expect(data).toBe('Error');
+          done();
+        }
+      });
+
+      actor.start();
+    });
+
+    // TODO: determine if tagged states should be used
+    it.skip('should complete (observer .complete)', (done) => {
+      const actor = interpret(fromPromise(() => Promise.resolve(42)));
+      expect.assertions(1);
+
+      actor.subscribe({
+        next: (state) => {
+          expect(state).toEqual(42);
+        },
+        complete: () => {
+          done();
+        }
+      });
+
+      actor.start();
+    });
+  });
+
+  describe('fromObservable', () => {
+    it('should resolve', (done) => {
+      const actor = interpret(fromObservable(() => of(42)));
+
+      actor.subscribe((state) => {
+        if (state === 42) {
+          done();
+        }
+      });
+
+      actor.start();
+    });
+
+    it('should resolve (observer .next)', (done) => {
+      const actor = interpret(fromObservable(() => of(42)));
+
+      actor.subscribe({
+        next: (state) => {
+          if (state === 42) {
+            done();
+          }
+        }
+      });
+
+      actor.start();
+    });
+
+    it('should reject (observer .error)', (done) => {
+      const actor = interpret(fromObservable(() => throwError(() => 'Error')));
+
+      actor.subscribe({
+        error: (data) => {
+          expect(data).toBe('Error');
+          done();
+        }
+      });
+
+      actor.start();
+    });
+
+    it('should complete (observer .complete)', (done) => {
+      const actor = interpret(fromObservable(() => EMPTY));
```

Comment:
Just an idea - `fromObservable` could probably accept observable directly. Observables can be hot but that's outside of our control anyway. The only thing we'd have to make sure about before making such a change is that `.subscribe` is called by us sufficiently late.